.github/workflows/smoke-tests.yml

```diff
@@ -0,0 +1,105 @@
+name: "System Smoke Tests"
+on:
+  push:
+    # run on pushes to master or release/*
+    branches:    
+      - master
+      - release/*
+  pull_request:
+    # run on pull requests targeting master
+    branches:
+      - master
+jobs:
+  setup:
+    runs-on: self-hosted
+    steps:
+    - name: Check Test Cluster
+      run: |
+        if kubectl get nodes; then
+          kubectl get nodes
+        else
+          echo "test cluster not active. Attempting to create through kind..."
+          kind delete cluster 2>/dev/null
+          kind create cluster
+          kubectl get nodes
+        fi
+  test-source:
+    runs-on: self-hosted
+    needs: setup
+    steps:
+    - uses: actions/checkout@v2
+    - name: Clean Cluster
+      run: |
+        kubectl delete all --all --wait=true --force --grace-period=0
+        sleep 5
+    - name: Deploy From Source
+      run: |
+        skaffold run
+    - name: Wait For Pods
+      run: |
+        sleep 60
+        kubectl get pods
+        echo waiting for ready state...
+        kubectl wait --for=condition=ready pod -l app=adservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=cartservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=checkoutservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=currencyservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=emailservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=frontend --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=loadgenerator --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=paymentservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=productcatalogservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=recommendationservice --timeout=500s
+        kubectl wait --for=condition=ready pod -l app=shippingservice --timeout=500s
+    - name: Test HTTP
+      timeout-minutes: 5
+      run: |
+        RESULT=" "
+        while [[ "$RESULT" != "  HTTP/1.1 200 OK" ]]; do
+          sleep 1
+          RESULT=$(kubectl exec deployments/frontend -- sh -c "wget --spider -S "http://frontend" 2>&1 | grep 'HTTP/'")
```

Comment:
since we're on localhost, don't we get an IP address to the frontend somehow with kind?

Reply:
I believe kind just shows "pending" for the external-ip field (https://github.com/kubernetes-sigs/kind/issues/702)

Reply:
yeah, but I suspect there's a nodePort for each Service that you can directly curl.
I bet kind has a way to query a Service from the linux machine where kind is installed? :)

Reply:
The research I've done is saying this is [more complicated than it looks](https://github.com/kubernetes-sigs/kind/issues/99#issuecomment-464614894), since we have to route through the extra layer of docker networking. If you're able to point me in the direction of a cleaner solution than this I'm happy to look into it though

Reply:
That's fine then no worries.

Reply:
if is a Linux machine, assuming you have access to the docker network and let's say one of the docker nodes has the IP 172.17.0.3, you can access services in a different way:
* a nodeport service, using one of the docker nodes ip address (i.e. 172.17.0.3:31788)
* a clusterIP service, adding a route to the service IP network or cluster IP address through one of the docker node IPs (i.e. ip route add 10.96.0.10 via 172.17.0.3)
* playing with the external-ip and LB as explained in the kind issue that you've mentioned

Reply:
I'd rather not add any new services or routes just for this test. I'd consider changing this test if there's a better one line option, but this so far seems like the least flaky option to me